front/pages/api/w/[wId]/apps/[aId]/runs/index.ts

```diff
@@ -111,19 +111,22 @@ async function handler(
       );
       const inputDataset = inputConfigEntry ? inputConfigEntry.dataset : null;
 
-      const dustRun = await coreAPI.createRun({
-        projectId: app.dustAPIProjectId,
-        runAsWorkspaceId: owner.sId,
-        runType: "local",
-        specification: dumpSpecification(
-          JSON.parse(req.body.specification),
-          latestDatasets
-        ),
-        datasetId: inputDataset,
-        config: { blocks: config },
-        credentials: credentialsFromProviders(providers),
-        secrets,
-      });
+      const dustRun = await coreAPI.createRun(
+        auth.getNonNullableWorkspace(),
```

Comment:
same here, we already have `owner` ?

Reply:
BTW here at the top we shouldn't use `getNonNullableWorkspace` because it throws while we want to 404 if we don't have a ws